sqlframe/base/dataframe.py

```diff
@@ -1852,9 +1852,10 @@ def head(self, n: int) -> t.List[Row]: ...
     def head(self, n: t.Optional[int] = None) -> t.Union[t.Optional[Row], t.List[Row]]:
         n = n or 1
         df = self.limit(n)
+        collected = df.collect()
         if n == 1:
-            return df.collect()[0]
-        return df.collect()
+            return None if collected == [] else collected[0]
```

Comment:
It looks like this method should only return a single element (so not a list) if no argument for head is provided. Therefore `.head()` behaves differently than `.head(1)`. https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.DataFrame.head.html

Can you fix that too while you are in here? I think you would need to record if n is None and check that at the end before overriding n to be 1. Therefore I think your line here where you check for 1 would change to check if None. 

Also if n is None, and therefore you want to return a single element or None you can use `seq_get(collected, n)`. `seq_get` is a helper from sqlglot.